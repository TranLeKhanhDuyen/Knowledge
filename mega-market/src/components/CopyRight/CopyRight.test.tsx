import { render } from '@testing-library/react'
import Copyright from '.'

describe('Copyright component', () => {
  test('renders with correct text content', () => {
    const companyName = 'Agility'
    const year = 2024
    const { getByText } = render(
      <Copyright companyName={companyName} year={year} />
    )

    expect(
      getByText(`© ${year} All rights reserved. ${companyName}.`)
    ).toBeInTheDocument()
  })

  test('renders with correct class names', () => {
    const companyName = 'Test Company'
    const year = 2024
    const { container } = render(
      <Copyright companyName={companyName} year={year} />
    )

    const paragraphElement = container.querySelector('p')
    expect(paragraphElement).toHaveClass('container')
    expect(paragraphElement).toHaveClass('copyright')
  })

  test('renders correctly with different props', () => {
    const companyName = 'Another Company'
    const year = 2025
    const { getByText } = render(
      <Copyright companyName={companyName} year={year} />
    )

    expect(
      getByText(`© ${year} All rights reserved. ${companyName}.`)
    ).toBeInTheDocument()
  })

  test('matches snapshot', () => {
    const companyName = 'Snapshot Company'
    const year = 2024
    const { asFragment } = render(
      <Copyright companyName={companyName} year={year} />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
