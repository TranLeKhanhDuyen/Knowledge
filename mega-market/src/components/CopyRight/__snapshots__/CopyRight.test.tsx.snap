// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copyright component matches snapshot 1`] = `
<DocumentFragment>
  <p
    class="container copyright"
  >
    Â© 2024 All rights reserved. Snapshot Company.
  </p>
</DocumentFragment>
`;
